### PR TITLE
feat: Add grid loss to basis data for energy calculations

### PR DESCRIPTION
**Pull request by BjarkeMeier**
# Description

Calculated grid loss is already being added to basis data for wholesale calculations.
This PR adds them as well for energy calculations.
In addition missing energy supplier has been added to positive and negative grid loss.

## Pull-request quality

- [x] The title adheres to [this guide](https://github.com/Mech0z/GitHubGuidelines)
- [x] Tests are written and executed locally
- [ ] Subsystem tests have been tested (by manually deploying to `dev_002` or `sandbox_002`)
- [ ] Documentation has been updated
- [ ] The integration [event catalog](https://energinet.atlassian.net/wiki/spaces/D3/pages/555581556/Event+catalog) has been updated
- [ ] C4 diagrams have been updated and Team Outlaws has been informed about relevant changes

### CODE REVIEW

File: source/databricks/calculation_engine/package/calculation/energy/aggregators/grid_loss_aggregators.py [hunk 1]
```diff
@@ -132,21 +134,24 @@ def calculate_negative_grid_loss(
...
-        f.lit(MeteringPointType.PRODUCTION.value).alias(Colname.metering_point_type),
-        Colname.qualities,
-        only_grid_area_and_metering_point_id[Colname.grid_loss_metering_point_id],
-    )
-
-    result = result.withColumnRenamed(
-        Colname.grid_loss_metering_point_id, Colname.metering_point_id
+    result = (
+        grid_loss.df.drop(Colname.energy_supplier_id)
+        .join(only_grid_area_and_metering_point_id, Colname.grid_area, "left")
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/package/calculation/energy/aggregators/grid_loss_aggregators.py:139]: isn't the variable name misleading now that it also contains energy supplier

Comment (JesperGomes) [reply source/databricks/calculation_engine/package/calculation/energy/aggregators/grid_loss_aggregators.py:139]: should you also join on energy supplier, and not drop it?

Comment (BjarkeMeier) [reply source/databricks/calculation_engine/package/calculation/energy/aggregators/grid_loss_aggregators.py]: Completely refactored.

File: source/databricks/calculation_engine/tests/calculation_logic/features/energy_calculations/flex_consumption_per_ga_and_es/when_period_is_one_day/test_data/metering_point_periods.csv [hunk 2]
```diff
@@ -1,4 +1,4 @@
 ﻿metering_point_id;type;calculation_type;settlement_method;grid_area_code;resolution;from_grid_area_code;to_grid_area_code;parent_metering_point_id;energy_supplier_id;balance_responsible_id;from_date;to_date
-170000000000000003;E17;;D01;804;PT15M;;;;1000000000000;1010000000000;2023-01-31 23:00;
-180000000000000002;E18;;;804;PT15M;;;;2000000000000;1020000000000;2023-01-31 23:00;
+170000000000000003;E17;;D01;804;PT15M;;;;1000000000000;;2023-01-31 23:00;
+180000000000000002;E18;;;804;PT15M;;;;2000000000000;;2023-01-31 23:00;
```
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/tests/calculation_logic/features/energy_calculations/flex_consumption_per_ga_and_es/when_period_is_one_day/test_data/metering_point_periods.csv:3]: For the reviewer: According to @khatozen grid loss metering points can not have a balance responsible party. So this is a bug fix of the test input data.
Feel free to close this comment when reviewing.

**Review by johevemi — APPROVED** [review]

**Pull request merged**